packages/pixelboy/apps/journal/checkin/checkin.coffee

```diff
@@ -47,3 +49,34 @@ class PAA.PixelBoy.Apps.Journal.CheckIn extends AM.Component
 
       else
         FlowRouter.go 'journal'
+
+  getExternalImgUrl: (event) ->
+    url = event.currentTarget.value
+    externalImgUrl = null
+    imgPreview = $('.image-preview')
+    placeholder = 'http://placehold.it/350x150?text=your+art'
+    errorIndicator = $('.preview-error')
+
+    if /twitter\.com/.test(url)
+      id = url.split('/status/')[1]
+      Meteor.call 'getImgFromTweet', id, (error, data) =>
+        if !error
+          externalImgUrl = data.entities.media[0].media_url_https
+          imgPreview.attr('src', externalImgUrl)
+          errorIndicator.empty()
+        else
+          imgPreview.attr('src', placeholder)
+          errorIndicator.html('There was an error communicating with the server. Either the tweet doesn\'t exist, or the server is down - try again later!')
+
+    else
+      imgPreview.attr('src', placeholder)
+      errorIndicator.html('Error! Can\'t parse that URL - try again!')
+
+  generateLocalImagePreview: (event) ->
+    if event.currentTarget.files && event.currentTarget.files[0]
```

Comment:
This can be more succinctly written using the `?` operator as: `if event.currentTarget.files?[0]`